nselib/url.lua

```diff
@@ -177,8 +178,22 @@ function parse(url, default)
     parsed.params = p
     return ""
   end)
+
   -- path is whatever was left
   parsed.path = url
+
+  -- Checks for folder route and extension
+  if(string.match(parsed.path, "/$")) then
+    parsed.is_folder = true
+  else
+    parsed.is_folder = false
+    local split_str = stdnse.strsplit('%.', parsed.path)
+    if(split_str and #split_str > 1) then
+      parsed.extension = stdnse.strsplit("%;", split_str[#split_str])[1]
+    end
```

Comment:
Thanks for the regex and explanation.
Fix committed as https://github.com/nmap/nmap/pull/952/commits/3d034e03629adc79e967189d1e9e4fa7f2ca080b.